src/index.spec.js

```diff
@@ -1,11 +1,192 @@
+const path = require('path')
+
+jest.doMock('./config')
+const config = require('./config')
+
+jest.doMock('./migrations')
+const migrations = require('./migrations')
+
+jest.doMock('./utils/fs')
+const fs = require('./utils/fs')
 
 const requireUncached = (moduleName) => {
-  delete require.cache[require.resolve(moduleName)]
-  return require(moduleName)
+  let mod
+  jest.isolateModules(() => {
+    mod = require(moduleName)
+  })
+  return mod
 }
 
+const main = require('./index')
+
+beforeEach(() => {
+  jest.resetAllMocks()
+})
+
 it('can be loaded without throwing exceptions', () => {
   expect(() => {
     requireUncached(`./index`)
   }).not.toThrow()
 })
+
+describe('init()', () => {
+  it('writes a configtemplate to the supplied path', async () => {
+    const targetPath = path.join(process.cwd(), 'exodus-init-test.js')
+    config.getSampleConfig.mockResolvedValueOnce('Hello world')
+
+    await main.init(targetPath)
+
+    expect(fs.writeFile).toHaveBeenCalledWith(targetPath, 'Hello world')
+  })
+})
+
+describe('create()', () => {
+  it('ensures the migrationsDirectory exists', async () => {
+    config.getConfig.mockResolvedValueOnce({
+      migrationsDirectory: '/dev/null',
+    })
+
+    await main.create().catch(() => {})
+
+    expect(fs.mkdir).toHaveBeenCalledWith(
+      '/dev/null',
+      expect.objectContaining({ recursive: true })
```

Comment:
I am just curious, is `expect.objectContaining` useful here?
Can't you just compare the whole object as second parameter?
i.e.
`
expect(fs.mkdir).toHaveBeenCalledWith(
  'dev/null',
  { recursive: true}
)
`

Reply:
You could use the object literal, but that would break if you appended additional options in the future. I don't know which approach is more desirable but figured the `mode` option _probably_ wont affect whether a directory is created or not. Filesystems arent my strongest area so I mostly winged it ☺️